packages/credential-provider-imds/src/fromInstanceMetadata.ts

```diff
@@ -25,10 +29,35 @@ export const fromInstanceMetadata = (init: RemoteProviderInit = {}): Provider<In
 const getInstanceImdsProvider = (init: RemoteProviderInit) => {
   // when set to true, metadata service will not fetch token
   let disableFetchToken = false;
+  const { logger, profile = process.env[AWS_PROFILE] ?? "default" } = init;
   const { timeout, maxRetries } = providerConfigFromInit(init);
 
   const getCredentials = async (maxRetries: number, options: RequestOptions) => {
-    const profile = (
+    const isImdsV1Fallback = disableFetchToken || options.headers?.["x-aws-ec2-metadata-token"] == null;
+
+    if (isImdsV1Fallback) {
+      const configFiles = await parseKnownFiles({});
+      const fallbackBlockedFromProfile =
+        configFiles[profile]?.ec2_metadata_v1_disabled &&
+        String(configFiles[profile]?.ec2_metadata_v1_disabled).toLowerCase() !== "false";
+      const fallbackBlockedFromProcessEnv =
+        process.env[AWS_EC2_METADATA_V1_DISABLED] && process.env[AWS_EC2_METADATA_V1_DISABLED] !== "false";
```

Comment:
Use configLoader from `node-config-provider`?
It memoizes the result. We just need to pass selectors.

https://github.com/awslabs/smithy-typescript/blob/39219d7d2377ed43f4fa9d827b9346e8a36a3a57/packages/node-config-provider/src/configLoader.ts